### ISSUE TITLE
Metals doesn't become aware of new build elements when bleep configuration changes. 

### ISSUE THREAD

**Issue by Baccata** [oyvindberg/bleep#274]
When I add a dependency to a project in bleep, I need to manually rerun the `metals: connect to build server` action metals to become aware of it.   

I've looked into how the bleep BSP is implemented, and from what I gather there doesn't seem to be anything that would regenerate the bloop config automatically upon the `bleep.yaml` changing, therefore the notification emitted by bloop to notify the BSP client (which is propagated by bleep to the IDE) is never sent. 

I personally think that the ideal developer experience would be comparable to the one of scala-cli (ie automatic and instantaneous awareness of changes in build configuration). 

Could we keep a file-watcher in the BSP implementation and re-create the bsp specific bloop configs upon it changing ?

**Comment by oyvindberg** [comment]
Absolutely. I tried when I implemented #221. 

However, the most naive thing somehow didn't work, so I decided to do it later and rather ship the PR quickly. 

The BSP integration in bleep is quite similar to that of scala-cli, and the file watch mode is copied/refactored from there, so hopefully it shouldn't be too hard to implement this. 

I really want it myself :)

**Comment by Baccata** [comment]
I'd be happy to give this a stab : I'd just need some guidance on how to set-up an environment that'd me point the `bsp` config to my local checkout of bleep. I have `bleep_dev` on my $PATH pointing to my local checkout, but I get the following when trying to run it to setup the IDE : 

``` 
📕 Error resolving artifact for bleep version 0.0.0: not found: https://github.com/oyvindberg/bleep/releases/download/v0.0.0/bleep-arm64-apple-darwin.tar.gz
```

**Comment by oyvindberg** [comment]
On the good side it is easy to set up, on the bad side the turnaround is long. The answer is to build a new native-image with the bleep script, and then point `.bsp/bleep.json` to the absolute path of that binary. Then you restart (restart, not reload/sync) the BSP server in your IDE.

I haven't had the time to work all that much with the BSP integration, and whatever I have done and copied have more or less worked fine the first time around. That's why it's not more developed.

I *think* the problem with pointing `.bsp/bleep.json` to `bleep_dev.sh` is that it produces a lot of garbage output to stdout. Maybe fixing it is as simple as adding a `--quiet` parameter to bleep itself.

We can also solve it by generating a launcher which starts a JVM and points it to the newest compiled class files. in fact this solution would be equivalent to what I implemented for the test harness in #247 , just saved as a shell script. This would replace `bleep_dev.sh`

There are other things I want to do to improve it as well, especially around tracing of BSP messages and testing. The interface of the java BSP integration library is so terrible that I have avoided spending too much time on it :)

**Comment by Baccata** [comment]
@keynmol and myself have a pet "ecosystem" that provides pure-scala LSP bindings, running the json-rpc bit on top of fs2. Anton has got some pretty cool UI bit to intercept the interaction and debug.

Anyway, thanks for the pointers, if I can sink some time into it I'll definitely try

I wonder if any of that could be extended to the bsp... Though I suppose the bloop launcher integrates tightly with the java bindings ...

**Comment by oyvindberg** [comment]
Ok, so this is fixed now - at least for metals, see #192 .
